sdks/common/fn-api/src/main/proto/beam_fn_api.proto

```diff
@@ -484,8 +484,24 @@ message StateRequest {
   // to state to be committed with the appropriate work execution.
   string instruction_reference = 2;
 
-  // (Required) The state key this request is for.
-  StateKey state_key = 3;
+  // (Required) The state key this request is for. The state key is always
+  // of the format [state class, class specific encoding]. The following
+  // classes are currently defined:
+  //   0-15    Reserved for Runner use
+  //   16      Multimap side input
+  //   17      User state with primary key
+  //   18      User state with primary and secondary key
+  //   19-inf  Reserved for future SDK usecases
+  //
+  // The class specific encodings are:
+  //   [int in [0, 15], runner specific opaque key]
+  //   [16, ptransform id, side input tag id, window, key]
```

Comment:
I think it would be helpful to document the types (aka coders) of the bits here. Is it:

 - Tag: `StringUtf8Coder`
 - Window: coder corresponding to XYZ windowingStrategy; (the bytes are opaque to the runner?)
 - Primary key & secondary key: opaque bytes; coder known by SDK, included in `StateSpec` and `SideInput` protos?